src/go/rpk/pkg/cli/cmd/group/seek.go

```diff
@@ -0,0 +1,268 @@
+		}
+
+		topic := fields[0]
+		partition, err := strconv.ParseInt(fields[1], 10, 32)
+		out.MaybeDie(err, "unable to parse partition on line %q: %v", line, err)
+		offset, err := strconv.ParseInt(fields[2], 10, 64)
+		out.MaybeDie(err, "unable to parse offset on line %q: %v", line, err)
+
+		o.Add(kadm.Offset{
+			Topic:       topic,
+			Partition:   int32(partition),
+			At:          offset,
+			LeaderEpoch: -1,
+		})
+	}
+	if err := s.Err(); err != nil {
+		out.Die("file %q scan error: %v", file, err)
+	}
+	return o
+}
+
+func seekFetch(
+	adm *kadm.Client, group string, topics map[string]bool,
+) kadm.Offsets {
+	resps, err := adm.FetchOffsets(context.Background(), group)
+	if err == nil {
+		err = resps.Error()
+	}
+	out.MaybeDie(err, "unable to fetch offsets for %q: %v", group, err)
+	if len(topics) > 0 {
+		resps.DeleteFunc(func(o kadm.OffsetResponse) bool { return !topics[o.Topic] })
+	}
+	return resps.Into()
+}
+
+func seek(
+	fs afero.Fs,
+	adm *kadm.Client,
+	group string,
+	to string,
+	toGroup string,
+	toFile string,
+	topics map[string]bool,
+	allowNewTopics bool,
+) {
+	current := seekFetch(adm, group, topics)
+	var commitTo kadm.Offsets
+	if toFile != "" {
+		commitTo = parseSeekFile(fs, toFile, topics)
+	} else if toGroup != "" {
+		commitTo = seekFetch(adm, toGroup, topics)
+	} else { // --to, we need to list offsets currently used, as well as any extra
+		tps := current.TopicsSet()
+		for topic := range topics {
+			if _, exists := tps[topic]; !exists && !allowNewTopics {
+				out.Die("Cannot commit new topic %q without --allow-new-topics.", topic)
+			}
+			tps[topic] = map[int32]struct{}{} // ensure exists
+		}
+		topics := tps.Topics()
+
+		var listed kadm.ListedOffsets
+		var err error
+		switch to {
+		case "start":
+			listed, err = adm.ListStartOffsets(context.Background(), topics...)
+		case "end":
+			listed, err = adm.ListEndOffsets(context.Background(), topics...)
+		default:
+			var milli int64
+			milli, err = strconv.ParseInt(to, 10, 64)
+			out.MaybeDie(err, "unable to parse millisecond %q: %v", to, err)
+			switch len(to) {
+			case len("1622505600"):
+				milli *= 1000 // sec to milli
+			case len("1622505600000"): // already in milli
+			case len("1622505600000000000"): // nano to milli
+				milli /= 1e6
+			default:
+				out.Die("--to timestamp %q is not a second, nor a millisecond, nor a nanosecond", to)
+			}
+			listed, err = adm.ListOffsetsAfterMilli(context.Background(), milli, topics...)
+		}
+		if err == nil { // ListOffsets can return ShardErrors, but we want to be entirely successful
+			err = listed.Error()
+		}
+		out.MaybeDie(err, "unable to list all offsets successfully: %v", err)
+		commitTo = listed.Into()
+	}
+
+	// Finally, we commit.
+	committed, err := adm.CommitOffsets(context.Background(), group, commitTo)
+	out.MaybeDie(err, "unable to commit offsets: %v", err)
+
+	useErr := committed.Error() != nil
+	headers := []string{"topic", "partition", "prior-offset", "current-offset"}
```

Comment:
Does it make it too confusing to read without the hyphen?

Reply:
I think it's fine without, but I think the strict column aspect of with is nice:
```
TOPIC  PARTITION  PRIOR-OFFSET  CURRENT-OFFSET
foo    0          10521         10440
foo    1          903           66
foo    2          571           332
```
vs
```
TOPIC  PARTITION  PRIOR OFFSET  CURRENT OFFSET
foo    0          10521         10440
foo    1          903           66
foo    2          571           332
```

Reply:
Also I just realized that `CURRENT-OFFSET` with the dash matches the output from `rpk group describe`, which was modeled after kafka-groups.sh.